src/Controller/ProfileController.php

```diff
@@ -15,24 +18,22 @@
 use Symfony\Component\Routing\Annotation\Route;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
 use Symfony\Component\HttpFoundation\File\UploadedFile;
-use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
 use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
-use Symfony\Component\Security\Core\User\UserInterface;
 
 class ProfileController extends AbstractController
 {
+
     /**
      * @Route("/profile", name="profile_index")
      * @IsGranted("ROLE_USER")
      */
-    public function profile(UserRepository $userRepository): Response
+    public function profile(): Response
     {
-        // Fetch UserEmail to get the property IsVerfied
-        if ($this->getUser() instanceof UserInterface) {
-            $userEmail = $this->getUser()->getUserIdentifier();
-            $user = $userRepository->findOneBy(['email' => $userEmail]);
+        // Fetch User to get the property IsVerfied
+        if ($this->getUser() instanceof User) {
+            $user = $this->getUser();
             // if it's not verified yet, we redirect the user to home with a flash message
-            if ($user === null || ($user !== null && !$user->getIsVerified())) {
+            if ($user == null || ($user !== null && !$user->getIsVerified())) {
```

Comment:
If $user is an instance of User it's not null

Reply:
Using lazy operators the right way can probably simplify this test and merge the previous one